lib/kinesis_client/stream/coordinator.ex

```diff
@@ -146,81 +155,89 @@ defmodule KinesisClient.Stream.Coordinator do
         :digraph.add_vertex(graph, shard_id)
       end
 
-      case s do
-        %{"ParentShardId" => parent_shard_id} when not is_nil(parent_shard_id) ->
-          add_vertex(graph, parent_shard_id)
-          :digraph.add_edge(graph, parent_shard_id, shard_id, "parent-child")
-
-          case s["AdjacentParentShardId"] do
-            nil ->
-              :ok
-
-            x when is_binary(x) ->
-              add_vertex(graph, x)
-              :digraph.add_edge(graph, x, shard_id)
-          end
-
-        _ ->
-          :ok
-      end
+      add_parent_shard(graph, s)
```

Comment:
This just cleans up the logic a little bit, no substantial changes to how we construct the graph.

Also note that this is one of the very few places that Erlang is essentially functionally impure: `graph` is a reference to stuff in ETS, so we don't return anything even though `graph` is in a sense being mutated.